src/components/progress-steps/progress-steps.jsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { cn } from '@/utilities/functions';
+import { Check } from 'lucide-react';
+import { getVariantClasses, completedStepCommonClasses, stepWrapperClasses } from './utils';
+
+/**
+ * ProgressSteps Component
+ *
+ * @param {Object}                    props               - Component props.
+ * @param {'dot' | 'number' | 'icon'} props.variant       - The type of step indicator.
+ * @param {'sm' | 'md' | 'lg'}        props.size          - The size of the step indicator.
+ * @param {'inline' | 'stack'}        props.type          - The layout type of the steps.
+ * @param {number}                    props.currentStep   - The current active step.
+ * @param {React.Element}             [props.variantIcon] - Custom icon for the 'icon' variant.
+ * @param {React.ReactNode}           props.children      - The steps to be rendered.
+ * @param {string}                    [props.className]   - Additional class names for the component.
+ */
+const ProgressSteps = ( {
+	variant = 'dot',
+	size = 'sm',
+	type = 'inline',
+	currentStep = 1,
+	variantIcon,
+	children,
+	className,
+	...props
+} ) => {
+	const totalSteps = React.Children.count( children );
+
+	const sizeClasses = {
+		sm: {
+			dot: 'size-2.5',
+			ring: 'size-5',
+			numberIcon: 'size-5 text-[10px]',
+			line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-2',
+			icon: 'size-5',
+		},
+		md: {
+			dot: 'size-3',
+			ring: 'size-6',
+			numberIcon: 'size-6 text-[12px]',
+			line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-3',
+			icon: 'size-6',
+		},
+		lg: {
+			dot: 'size-3.5',
+			ring: 'size-7',
+			numberIcon: 'size-7 text-[14px]',
+			line: type === 'stack' ? 'h-0.5' : 'h-0.5 mx-2',
+			icon: 'size-7',
+		},
+	};
+
+	const steps = React.Children.toArray( children ).map( ( child, index ) => {
+		const isCompleted = index + 1 < currentStep;
+		const isCurrent = index + 1 === currentStep;
+
+		const stepContent = createStepContent( variant, isCompleted, isCurrent, sizeClasses, size, index, variantIcon );
+
+		let stepRingClass;
+
+		if ( isCurrent ) {
+			stepRingClass = 'ring-1 ring-brand-primary-600';
+		} else if ( isCompleted ) {
+			stepRingClass = ''; // No ring for completed steps (based on the original logic)
+		} else {
+			stepRingClass = 'ring-1 ring-gray-400';
+		}
+
+		const stepClasses = cn(
+			'relative rounded-full flex items-center justify-center transition-colors duration-500',
+			stepRingClass,
+			sizeClasses[ size ].ring
+		);
+
+		return (
+			<React.Fragment key={ index }>
+				{ React.cloneElement( child, {
+					stepContent,
+					stepClasses,
+					isCurrent,
+					isCompleted,
+					totalSteps,
+					type,
+				} ) }
+				{ index < totalSteps - 1 && (
+					<div
+						className={ `flex-1 ${ sizeClasses[ size ].line } transition-colors duration-500 ${
+							( isCompleted ) ? 'bg-brand-primary-600' : 'bg-gray-300'
+						}` }
```

Comment:
For conditional class names use `cn` function and try to avoid `Template strings`.